livechat/tests/test_billing_api_client.py

```diff
@@ -0,0 +1,92 @@
+
+def test_get_client_without_access_token():
+    ''' Test if TypeError raised without access_token. '''
+    with pytest.raises(TypeError) as exception:
+        BillingApi.get_client(version='test')
+    assert str(
+        exception.value
+    ) == "get_client() missing 1 required positional argument: 'token'"
+
+
+def test_get_client_with_non_existing_version():
+    ''' Test if ValueError raised for non-existing version. '''
+    with pytest.raises(ValueError) as exception:
+        BillingApi.get_client(token='test', version='test')
+    assert str(exception.value) == 'Provided version does not exist.'
+
+
+def test_get_client_with_valid_args(billing_api_client):
+    ''' Test if production API URL is used and token is added to headers for valid args. '''
+    assert billing_api_client.api_url == 'https://billing.livechatinc.com/v1'
+    assert billing_api_client.session.headers.get('Authorization') == 'test'
+
+
+def test_send_request(billing_api_client):
+    ''' Test if it's possible to send a basic request via Billing API
+        client with arbitrary chosen method. '''
+    assert billing_api_client.create_direct_charge().json() == {
+        'error':
+        'invalid_request',
+        'error_description':
+        'The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed.'
+    }
+
+
+def test_modify_header(billing_api_client):
+    ''' Test if Billing API object header can be updated with custom value. '''
+    assert 'test' not in billing_api_client.get_headers()
+    billing_api_client.modify_header({'test': '1234'})
+    assert 'test' in billing_api_client.get_headers()
+
+
+def test_remove_header(billing_api_client):
+    ''' Test if header can be removed from Billing API object. '''
+    billing_api_client.modify_header({'test2': '1234'})
+    assert 'test2' in billing_api_client.get_headers()
+    billing_api_client.remove_header('test2')
+    assert 'test2' not in billing_api_client.get_headers()
+
+
+def test_custom_headers_within_the_request(billing_api_client):
+    ''' Test if custom headers can be added to the session headers
+        only within the particular request. '''
+    headers = {'x-test': 'enabled'}
+    response = billing_api_client.create_direct_charge(headers=headers)
+    assert headers.items() <= response.request.headers.items()
+    assert 'x-test' not in billing_api_client.get_headers()
+
+
+def test_client_supports_http_1():
+    ''' Test if client supports HTTP/1.1 protocol. '''
+    client = BillingApi.get_client(token='test')
+    assert client.create_direct_charge().http_version == 'HTTP/1.1'
+
+
+# At this point billing-api v1 support http2 but will always negotiate to use 'HTTP/1.1'
+# def test_client_supports_http_2():
+#     ''' Test if client supports HTTP/2 protocol. '''
+#     client = BillingApi.get_client(token='test', http2=True)
+#     assert client.create_direct_charge().http_version == 'HTTP/2'
```

Comment:
Not sure what it means but don't leave commented code. Delete these lines if YAGNI.